File: converted/source2/source/zombie/core/Rand$PZSeedGenerator.java
package zombie.core;

import org.uncommons.maths.random.SecureRandomSeedGenerator;
import org.uncommons.maths.random.SeedException;
import org.uncommons.maths.random.SeedGenerator;

public final class Rand$PZSeedGenerator implements SeedGenerator {
   private static final SeedGenerator[] GENERATORS = new SeedGenerator[]{new SecureRandomSeedGenerator()};

   private Rand$PZSeedGenerator() {
   }

   public byte[] generateSeed(int var1) {
      for (SeedGenerator var5 : GENERATORS) {
         try {
            return var5.generateSeed(var1);
         } catch (SeedException var7) {
         }
      }

      throw new IllegalStateException("All available seed generation strategies failed.");
   }
}

File: DoNotUse/include/zombie/core/Rand$PZSeedGenerator.cpp
#pragma once
#include <string>
#include <vector>
#include <memory>
#include <unordered_map>
#include <unordered_set>
#include <cstdint>
#include <random>

namespace zombie {
namespace core {


class Rand {
public:
   private static const SeedGenerator[] GENERATORS = new SeedGenerator[]{std::make_unique<SecureRandomSeedGenerator>()};

   private Rand$PZSeedGenerator() {
   }

   public byte[] generateSeed(int var1) {
    for (auto& var5 : GENERATORS)         try {
            return var5.generateSeed(var1);
         } catch (SeedException var7) {
         }
      }

      throw IllegalStateException("All available seed generation strategies failed.");
   }
}
} // namespace core
} // namespace zombie
